File: testers/combined/java/generated-and-checked-in/avro_events/combined/avro_events/consumer/InvoiceHandler.java
package combined.avro_events.consumer;

import com.example.events.Invoice;
import combined.avro_events.header.StandardHeaders;
import java.lang.Void;
import java.util.concurrent.CompletableFuture;

/** Handler interface for invoice topic events */
public interface InvoiceHandler {
  /** Handle a message from the topic */
  CompletableFuture<Void> handle(
    String key,
    Invoice value,
    StandardHeaders headers
  );
}

File: testers/combined/kotlin/generated-and-checked-in/avro_events/combined/avro_events/consumer/InvoiceHandler.kt
package combined.avro_events.consumer

import com.example.events.Invoice
import combined.avro_events.header.StandardHeaders
import io.smallrye.mutiny.Uni

/** Handler interface for invoice topic events */
interface InvoiceHandler {
  /** Handle a message from the topic */
  abstract fun handle(
    key: kotlin.String,
    value: Invoice,
    headers: StandardHeaders
  ): Uni<Unit>
}